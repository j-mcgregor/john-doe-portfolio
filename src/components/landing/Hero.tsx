import React from 'react'
import { HeroProps, StyledHeroProps } from '../../types/interfaces'
import styled from 'styled-components'
import { Breakpoints } from '../../types/enums'

const StyledHero = styled.div<StyledHeroProps>`
    background: ${({ bgImageUrl }) =>
        bgImageUrl ? `url(${bgImageUrl})` : 'black'};
    background-size: cover;
    background-position: right center;
    background-repeat: no-repeat;
    background-attachment: fixed;
    height: 700px;
    text-transform: uppercase;

    h3 {
        font-family: 'Abel', sans-serif;
    }

    @media screen and (max-width: ${Breakpoints.md}px) {
        height: 600px;
    }
    @media screen and (max-width: ${Breakpoints.sm}px) {
        height: 500px;
    }
`

const Hero: React.FC<HeroProps> = ({
    primary,
    secondary,
    logo,
    backgroundImage,
}) => {
    return (
        <StyledHero
            bgImageUrl={backgroundImage}
            className="flex flex-column flex-center"
        >
            {logo || null}
            {primary}
            {secondary}
        </StyledHero>
    )
}

export default Hero
