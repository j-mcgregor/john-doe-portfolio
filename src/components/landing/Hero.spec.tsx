import React from 'react'
import renderer from 'react-test-renderer'
import Img from 'gatsby-image'
import Hero from './Hero'

describe('Hero', () => {
    it('should render with no logo', () => {
        // Assemble
        const primary = <h2>Primary Text</h2>
        const secondary = <h3>Secondary Text</h3>
        const hero = renderer.create(
            <Hero primary={primary} secondary={secondary} />
        )
        const heroInstance = hero.toJSON()

        // Assert
        expect(heroInstance).not.toBeNull()
        if (heroInstance) {
            expect(heroInstance).toMatchSnapshot()
        }
    })

    it('should render with logo if provided', () => {
        // Assemble
        const primary = <h2>Primary Text</h2>
        const secondary = <h3>Secondary Text</h3>

        const imgData = {
            base64: 'base:image/png',
            width: 100,
            height: 100,
            src: 'logo.png',
            srcSet: 'logo.png',
        }
        const logo = <Img fixed={imgData} alt="Logo" />

        const hero = renderer.create(
            <Hero primary={primary} secondary={secondary} logo={logo} />
        )
        const heroInstance = hero.toJSON()

        // Assert
        expect(heroInstance).not.toBeNull()
        if (heroInstance) {
            expect(heroInstance).toMatchSnapshot()
        }
    })

    it('should pass down a background image if provided', () => {
        // Assemble
        const primary = <h2>Primary Text</h2>
        const secondary = <h3>Secondary Text</h3>
        const backgroundImage =
            'https://images.prismic.io/john-doe-portfolio/test.png'

        const hero = renderer.create(
            <Hero
                primary={primary}
                secondary={secondary}
                backgroundImage={backgroundImage}
            />
        )
        const heroInstance = hero.toJSON()

        // Assert
        expect(heroInstance).not.toBeNull()
        if (heroInstance) {
            expect(heroInstance).toMatchSnapshot()
        }
    })
})
