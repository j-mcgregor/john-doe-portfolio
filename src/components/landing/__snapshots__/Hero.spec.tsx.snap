// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero should pass down a background image if provided 1`] = `
<div
  className="sc-AxjAm hJfqat flex flex-column flex-center"
>
  <h2>
    Primary Text
  </h2>
  <h3>
    Secondary Text
  </h3>
</div>
`;

exports[`Hero should render with logo if provided 1`] = `
<div
  className="sc-AxjAm cTHBye flex flex-column flex-center"
>
  <div
    className=" gatsby-image-wrapper"
    style={
      Object {
        "display": "inline-block",
        "height": 100,
        "overflow": "hidden",
        "position": "relative",
        "width": 100,
      }
    }
  >
    <img
      alt=""
      aria-hidden={true}
      src="base:image/png"
      style={
        Object {
          "height": "100%",
          "left": 0,
          "objectFit": "cover",
          "objectPosition": "center",
          "opacity": 1,
          "position": "absolute",
          "top": 0,
          "transitionDelay": "500ms",
          "width": "100%",
        }
      }
    />
    <picture>
      <source
        srcSet="logo.png"
      />
      <img
        alt="Logo"
        height={100}
        loading="lazy"
        onLoad={[Function]}
        src="logo.png"
        srcSet="logo.png"
        style={
          Object {
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "transition": "opacity 500ms",
            "width": "100%",
          }
        }
        width={100}
      />
    </picture>
    <noscript
      dangerouslySetInnerHTML={
        Object {
          "__html": "<picture><source srcset=\\"logo.png\\" /><img loading=\\"lazy\\" width=\\"100\\" height=\\"100\\" srcset=\\"logo.png\\" src=\\"logo.png\\" alt=\\"Logo\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
        }
      }
    />
  </div>
  <h2>
    Primary Text
  </h2>
  <h3>
    Secondary Text
  </h3>
</div>
`;

exports[`Hero should render with no logo 1`] = `
<div
  className="sc-AxjAm cTHBye flex flex-column flex-center"
>
  <h2>
    Primary Text
  </h2>
  <h3>
    Secondary Text
  </h3>
</div>
`;
